pkg/client/group_mgmt_client.go

```diff
@@ -44,20 +47,27 @@ type ErrorResponse struct {
 
 // Message is an `Error` implementation as well as an implementation of the JSON API error object.
 type Message struct {
-	Code string `json:"code,omitempty"`
-	Text string `json:"text,omitempty"`
+	Code      string   `json:"code,omitempty"`
+	Text      string   `json:"text,omitempty"`
+	Severity  string   `json:"severity,omitempty"`
+	Arguments Argument `json:"arguments,omitempty"`
+}
+
+type Argument struct {
+	JobId string `json:"job_id,omitempty"`
 }
 
 // NewClient instantiates a new client to communicate with the Nimble group
-func NewClient(ipAddress, username, password, apiVersion string) (*GroupMgmtClient, error) {
+func NewClient(ipAddress, username, password, apiVersion string, waitOnJobs bool) (*GroupMgmtClient, error) {
```

Comment:
Sorry for the late comment here.  Can we change the constructor to use the functional options pattern: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Thanks

Reply:
@rgcostea , can we take this as an enhancement and close this pull request?